Ft domain skeleton

SukafuDesu opened the pull request.

SukafuDesu committed:
[Feature] Created Domain Skeleton

* Created Interface for Repositories
* Created Domain Models
* Added Documentation for Interfaces

SukafuDesu committed:
Reduced Code Smells

* Added Async Suffix to the end
* Removed Comment about code

mendac017 commented on FakebookNotifications/FakebookNotifications.Domain/Models/User.cs:
```diff
@@ -0,0 +1,50 @@
+        /// </returns>
+        public bool IsValid()
+        {
+            return !(string.IsNullOrEmpty(Id) || string.IsNullOrEmpty(Email));
```
We don't want implementation in this PR. Instead have it throw not implemented for all methods.

mendac017 reviewed (changes requested):
There should be tests for domain methods that fail.

Commit:
Merge branch 'us-follow-notification' into ft-domain-skeleton

ghost commented:
The docker build also failed in the pipeline.

SukafuDesu committed:
* replaced function functionality with throw not implemented()

SukafuDesu committed:
Merge branch 'ft-domain-skeleton' of https://github.com/2011-fakebook-project3/notifications into ft-domain-skeleton

SukafuDesu committed:
Merge branch 'us-follow-notification' into ft-domain-skeleton

SukafuDesu committed:
Merge branch 'us-follow-notification' into ft-domain-skeleton

SukafuDesu committed:
Created Test for Domain User

* Added UserIsValidTrue
* Added UserIsValidFalse
* Updated Docker to have Domain Class

mendac017 reviewed (approved):
This looks good. I tested it and the tests do fail.

townerr commented on FakebookNotifications/FakebookNotifications.Domain/Models/User.cs:
```diff
@@ -0,0 +1,50 @@
+                    _id = "";
+                }
+            }
+                
```
fix brace formatting

townerr commented on FakebookNotifications/FakebookNotifications.Testing/FakebookNotifications.Testing.csproj:
```diff
@@ -23,6 +23,7 @@
 	<ItemGroup>
```
item group tabbed in too far

townerr commented on FakebookNotifications/FakebookNotifications.sln:
```diff
@@ -6,6 +6,8 @@ MinimumVisualStudioVersion = 10.0.40219.1
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "FakebookNotifications.Testing", "FakebookNotifications.Testing\FakebookNotifications.Testing.csproj", "{3D40AD3A-1E08-4135-867A-AAD78677079D}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "FakebookNotifications.Domain", "FakebookNotifications.Domain\FakebookNotifications.Domain.csproj", "{10727CC8-E023-4F66-8510-151B65218FD0}"
```
end project for this

SukafuDesu committed:
Fixed Indentation

• updated User.cs with better indentation.
• added lambda statement to Id String
• fixed indentation on csproj for item group

SukafuDesu committed:
Merge branch 'us-follow-notification' into ft-domain-skeleton
